zanata-war/src/main/java/org/zanata/limits/RateLimitingProcessor.java

```diff
@@ -36,6 +36,16 @@ public void processApiKey(String apiKey, HttpResponse response,
         process(apiKey, response, taskToRun);
     }
 
+    public void processUsername(String username, HttpResponse response,
+        Runnable taskToRun) throws IOException {
+        process(username, response, taskToRun);
+    }
+
+    public void processClientIp(String ip, HttpResponse response,
+        Runnable taskToRun) throws IOException {
+        process(ip, response, taskToRun);
+    }
+
     private void process(String key, HttpResponse response, Runnable taskToRun)
```

Comment:
"process" is something we do to the `Runnable` task, so processUsername and processClientIp don't quite make sense.  (I know, processUsername already existed.)

What about `processForUser` and `processForAnonymousIP`?